### PR TITLE
[FEAT] TT-1138 Remove CDIS Dependencies

### PR DESCRIPTION
**Pull request by huarngpa**
* Move parcel over
* Add tox
* Make Python3 compatible

### CODE REVIEW

File: gdc_client/parcel/client.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
+from . import const
```
Comment (anmaxvl) [inline gdc_client/parcel/client.py]: how copy-paste is this file? if most of it is, please, include the link or reference to the original author.

File: gdc_client/parcel/defaults.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+tcp_url = 'https://localhost'
+udt_url = 'https://localhost'
```
Comment (anmaxvl) [inline gdc_client/parcel/defaults.py]: 💣

File: tests/test_query.py [hunk 3]
```diff
@@ -32,31 +32,31 @@ def tearDown(self):
     def test_no_metadata_get_related_files(self):
         index = GDCIndexClient(uri=base_url)
 
-        results = index.get_related_files(uuids['small'])
+        results = index.get_related_files('small')
```
Comment (ProfOak) [inline tests/test_query.py:35]: Can you elaborate on this change? I think this method accepts a list of uuids.

File: gdc_client/download/client.py [hunk 4]
```diff
@@ -320,24 +326,3 @@ def parallel_download(self, stream):
                     file_id, e))
                 if self.debug:
                     raise
-
-
-class GDCUDTDownloadClient(GDCHTTPDownloadClient):
```
Comment (ProfOak) [inline gdc_client/download/client.py:325]: This is probably ok. I don't think there are plans to use UDT in the near future.

**Review by ProfOak — COMMENTED** [review]
Seems reasonable but I had one question.

File: gdc_client/download/client.py (hunk 4 above)
Comment (huarngpa) [reply gdc_client/download/client.py:325]: Yeah, just remove it :(

File: tests/test_query.py (hunk 3 above)
Comment (huarngpa) [reply tests/test_query.py:35]: `uuids['small']` is a [dict](https://github.com/NCI-GDC/gdc-client/blob/a7f147c5ad4a60dc68784de7e3a7cc48f3f2bde6/tests/conftest.py#L46). I think in python2 this was fine (possibly dict types could be dictionary keys)?? But python3 doesn't allow this. See this [line](https://github.com/NCI-GDC/gdc-client/blob/a7f147c5ad4a60dc68784de7e3a7cc48f3f2bde6/gdc_client/query/index.py#L19) for what I mean... it expects a type string and then searches through a list of dictkeys

Comment (ProfOak) [reply tests/test_query.py:35]: Ah that makes sense. My mistake.

**Review by ProfOak — APPROVED** [review]

File: gdc_client/parcel/download_stream.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
+from . import utils
```
Comment (anmaxvl) [inline gdc_client/parcel/download_stream.py]: same here

File: gdc_client/parcel/client.py [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+                        raise
+                    else:
+                        log.error("Download aborted: {0}".format(str(e)))
+                        break
+
+        # Divide work amongst process pool
+        pool = [Process(target=download_worker) for i in range(n_procs)]
+
+        # Start pool
+        map(lambda p: p.start(), pool)
```
Comment (anmaxvl) [inline gdc_client/parcel/client.py]: not gonna work in py3. and a pretty bad practice in general to use the side-effect behavior 🤷‍♂

File: gdc_client/parcel/log.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import logging
+import sys
+from .portability import colored
+
+
+loggers = {}
+
+
+# Logging
+def get_logger(name='parcel'):
```
Comment (anmaxvl) [inline gdc_client/parcel/log.py:19]: looks familiar 🤔 :trollface:

File: gdc_client/parcel/segment.py [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+        self.pbar = None
+        self.pbar = get_pbar(self.download.url, self.download.size)
+
+    def _setup_work(self):
+        if self.is_complete():
+            log.debug('File already complete.')
+            return
+
+        work_size = self.integrate(self.work_pool)
+        self.block_size = work_size / self.n_procs
```
Comment (anmaxvl) [inline gdc_client/parcel/segment.py]: `//` to avoid float

File: gdc_client/parcel/utils.py [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+def check_transfer_size(actual, expected):
+    """Simple validation on any expected versus actual sizes.
+
+    :param int actual: The size that was actually transferred
+    :param int actual: The size that was expected to be transferred
+
+    """
+
+    return actual == expected
```
Comment (anmaxvl) [inline gdc_client/parcel/utils.py:39]: hm, ok 🤷‍♂

File: gdc_client/parcel/utils.py [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+        s=' {0} '.format(file_id), n=50, c='-')))
+
+
+def print_closing_header(file_id):
+    log.debug('^{0}^'.format('{s:{c}^{n}}'.format(
+        s=' {0} '.format(file_id), n=50, c='-')))
+
+def write_offset(path, data, offset):
+    try:
+        f = open(path, 'r+b')
```
Comment (anmaxvl) [inline gdc_client/parcel/utils.py]: mb do `with open(path, 'r+b') as f:` instead?

File: gdc_client/parcel/utils.py [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+        f.seek(offset)
+        f.write(data)
+        f.close()
+    except Exception as e:
+        raise Exception('Unable to write offset: {0}'.format(str(e)))
+
+
+def read_offset(path, offset, size):
+    try:
+        f = open(path, 'r+b')
```
Comment (anmaxvl) [inline gdc_client/parcel/utils.py]: same as above

File: gdc_client/parcel/utils.py [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+        return data
+    except Exception as e:
+        raise Exception('Unable to read offset: {0}'.format(str(e)))
+
+
+def set_file_length(path, length):
+    try:
+        if os.path.isfile(path) and os.path.getsize(path) == length:
+            return
+        f = open(path, 'wb')
```
Comment (anmaxvl) [inline gdc_client/parcel/utils.py]: same

**Review by anmaxvl — CHANGES_REQUESTED** [review]

File: gdc_client/parcel/defaults.py (hunk 2 above)
Comment (huarngpa) [reply gdc_client/parcel/defaults.py]: gone

File: gdc_client/parcel/client.py (hunk 1 above)
Comment (huarngpa) [reply gdc_client/parcel/client.py]: good point!

File: gdc_client/parcel/log.py (hunk 7 above)
Comment (huarngpa) [reply gdc_client/parcel/log.py:19]: lol

File: gdc_client/parcel/utils.py (hunk 9 above)
Comment (huarngpa) [reply gdc_client/parcel/utils.py:39]: lol.. let's just move on

File: gdc_client/parcel/utils.py (hunk 10 above)
Comment (huarngpa) [reply gdc_client/parcel/utils.py]: good point

File: gdc_client/parcel/client.py (hunk 6 above)
Comment (huarngpa) [reply gdc_client/parcel/client.py]: thanks for the catch!

**Review by anmaxvl — APPROVED** [review]
lgtm

File: gdc_client/download/client.py [hunk 13]
```diff
@@ -5,12 +5,17 @@
 import sys
 import tarfile
 import time
-import urlparse
-from StringIO import StringIO
+from six.moves.urllib.parse import urljoin
```
Comment (kulgan) [inline gdc_client/download/client.py]: Can you do an as instead
``` python
from six.moves.urllib.parse as urlparse
```

File: gdc_client/parcel/__init__.py [hunk 14]
```diff
@@ -0,0 +1,2 @@
+from .http_client import HTTPClient
```
Comment (kulgan) [inline gdc_client/parcel/__init__.py]: Can you avoid relative imports

File: gdc_client/download/client.py (hunk 13 above)
Comment (huarngpa) [reply gdc_client/download/client.py]: ah, clever :+1:

File: gdc_client/parcel/__init__.py (hunk 14 above)
Comment (huarngpa) [reply gdc_client/parcel/__init__.py]: Fixing

**Pull request merged**